src/CommitmentsV2/SFA.DAS.CommitmentsV2/Application/Commands/DeleteDraftApprenticeship/DeleteDraftApprenticeshipHandler.cs

```diff
@@ -0,0 +1,44 @@
+﻿using MediatR;
+using Microsoft.Extensions.Logging;
+using SFA.DAS.CommitmentsV2.Data;
+using SFA.DAS.CommitmentsV2.Domain.Interfaces;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.CommitmentsV2.Application.Commands.DeleteDraftApprenticeship
+{
+    public class DeleteDraftApprenticeshipHandler : IRequestHandler<DeleteDraftApprenticeshipCommand, DeleteDraftApprenticeshipResponse>
+    {
+        private readonly Lazy<ProviderCommitmentsDbContext> _dbContext;
+        private readonly ILogger<DeleteDraftApprenticeshipHandler> _logger;
+        private readonly ICohortDomainService _cohortDomainService;
+
+        public DeleteDraftApprenticeshipHandler(
+            Lazy<ProviderCommitmentsDbContext> dbContext,
+            ILogger<DeleteDraftApprenticeshipHandler> logger,
+            ICohortDomainService cohortDomainService)
+        {
+            _dbContext = dbContext;
+            _logger = logger;
+            _cohortDomainService = cohortDomainService;
+        }
+
+        public async Task<DeleteDraftApprenticeshipResponse> Handle(DeleteDraftApprenticeshipCommand command, CancellationToken cancellationToken)
+        {
+            await _cohortDomainService.DeleteDraftApprenticeship(command.CohortId, command.ApprenticeshipId, command.UserInfo, cancellationToken);
+
+            _logger.LogInformation($"Deleted apprenticeShip. Apprenticeship-Id:{command.ApprenticeshipId} Commitment-Id:{command.CohortId}");
+
+            var response = new DeleteDraftApprenticeshipResponse
+            {
+                Id = command.CohortId,
+                ApprenticeshipId = command.ApprenticeshipId
+            };
+
+            return response;
+        }
```

Comment:
I don't think this command requires any response. We are just returning what we sent in.

Reply:
Since we have a logger in-place, it might be useful to log any exceptions which occur whilst trying to delete.

Reply:
Agreed, I will remove the response and the class.